app/web/src/components/Connection.vue

```diff
@@ -73,6 +118,30 @@ const componentsStore = useComponentsStore();
 const viewsStore = useViewsStore();
 const modelingEventBus = componentsStore.eventBus;
 
+const componentNames = computed(() => {
+  if (isSubscriptionConnection(props.connection)) {
+    const toComponent =
+      componentsStore.allComponentsById[props.connection.toComponentId];
+    const fromComponent =
+      componentsStore.allComponentsById[props.connection.fromComponentId];
+    return {
+      to: toComponent?.def.displayName ?? "?",
+      from: fromComponent?.def.displayName ?? "?",
+    };
+  }
+  if (isSocketConnection(props.connection)) {
```

Comment:
Also a very good place to use if / else, so that typescript will essentially do exhaustive matching!